tests/utils.py

```diff
@@ -1018,8 +1018,16 @@ def snapshot_context(
         except Exception as e:
             pytest.fail("Could not connect to test agent: %s" % str(e), pytrace=False)
         else:
-            r = conn.getresponse()
-            if r.status != 200:
+            r = None
+            attempt_start = time.time()
+            while r is None and time.time() - attempt_start < 60:
+                try:
+                    r = conn.getresponse()
```

Comment:
worth adding a sleep? I know you are checking for up to 60 seconds, but we don't need to hammer the service, half a second pause between checks?